### PR TITLE
Inject drawerLayout in TrendingView. Open drawer by clicking navigation icon.

### LINKED ISSUE #126: Open drawer by clicking toolbar navigation icon
From `TrendingView.java`:
```java
  toolbarView.setNavigationOnClickListener(new OnClickListener() {
      @Override public void onClick(View v) {
        // TODO bind to drawer with... injection?
      }
    });
```

How are you guys planning to implement this binding with `MainActivity.drawerLayout`?

Typically I'm doing this the following way:
`ActivityConnector` binds to Activity livecycle with attach/detach (e.g. called from onStart/onStop), holds reference to some object in Activity. Simply it looks like this:
```java
public class ActivityConnector<T> {
  private WeakReference<T> ref;

  public void attach(T obj) {
    ref = new WeakReference<T>(obj);
  }

  public void detach() {
    ref = null;
  }

  public T get() {
    return ref == null ? null : ref.get();
  }
}
```
More robust implementation can be found [here](https://github.com/LiveTyping/u2020-mvp/blob/master/app/src/main/java/ru/ltst/u2020mvp/base/ActivityConnector.java).

This approach can be used to bind `drawerLayout`:
```java
public final class MainActivity extends Activity {
  @Inject ActivityConnector<DrawerLayout> drawerConnector;

  @InjectView(R.id.main_drawer_layout) DrawerLayout drawerLayout;

  @Override protected void onStart() {
    super.onStart();
    drawerConnector.attach(drawerLayout);
  }

  @Override protected void onStop() {
    super.onStop();
    drawerConnector.detach();
  }
}
```
in `TrendingView.java`:
```java
@Inject ActivityConnector<DrawerLayout> drawerConnector;
@InjectView(R.id.trending_toolbar) Toolbar toolbarView;

@Override protected void onFinishInflate() {
   super.onFinishInflate();
   ButterKnife.inject(this);

   toolbarView.setNavigationIcon(R.drawable.menu_icon);
   toolbarView.setNavigationOnClickListener(new OnClickListener() {
     @Override public void onClick(View v) {
       DrawerLayout drawer = drawerConnector.get();
       if (drawer != null) {
         drawerLayout.openDrawer(Gravity.START);
       }
     }
   });
}
```

What do you think about that?

### PR DESCRIPTION
**Pull request by kboyarshinov**
Closes #126.

### CODE REVIEW

**Comment by JakeWharton** [comment]
Thanks!
